examples/cis2-wccd-upgrade/src/lib.rs

```diff
@@ -0,0 +1,2030 @@
+    };
+    Ok(())
+}
+
+/// Set unpause_time.
+#[receive(contract = "CIS2-wCCD-State", name = "set_unpause_time", mutable)]
+fn contract_state_set_unpause_time<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    // Only implementation can set state.
+    only_implementation(host.state().implementation_address, ctx.sender())?;
+
+    // Set unpause_time.
+    let unpause_time = ctx.parameter_cursor().get()?;
+    host.state_mut().unpause_time = unpause_time;
+    Ok(())
+}
+
+#[receive(contract = "CIS2-wCCD-State", name = "get_unpause_time", return_value = "UnpauseTime")]
+fn contract_state_get_unpause_time<S: HasStateApi>(
+    _ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<Timestamp> {
+    Ok(host.state().unpause_time)
+}
+
+/// Initialize contract instance with no initial tokens.
+/// Logs a `Mint` event for the single token id with no amounts.
+/// TODO: Move event logs to proxy
+#[init(contract = "CIS2-wCCD", enable_logger)]
+fn contract_init<S: HasStateApi>(
+    ctx: &impl HasInitContext,
+    state_builder: &mut StateBuilder<S>,
+    logger: &mut impl HasLogger,
+) -> InitResult<StateImplementation<S>> {
+    // Get the instantiater of this contract instance.
+    let invoker = Address::Account(ctx.init_origin());
+    // Construct the initial contract state.
+    let state = StateImplementation::new(invoker, state_builder);
+    // Log event for the newly minted token.
+    logger.log(&Cis2Event::Mint(MintEvent {
+        token_id: TOKEN_ID_WCCD,
+        amount:   ContractTokenAmount::from(0u64),
+        owner:    invoker,
+    }))?;
+
+    // Log event for where to find metadata for the token
+    logger.log(&Cis2Event::TokenMetadata::<_, ContractTokenAmount>(TokenMetadataEvent {
+        token_id:     TOKEN_ID_WCCD,
+        metadata_url: MetadataUrl {
+            url:  String::from(TOKEN_METADATA_URL),
+            hash: None,
+        },
+    }))?;
+
+    Ok(state)
+}
+
+/// Initialzes the implementation of the w_ccd contract with the proxy and the
+/// state addresses.
+#[receive(
+    contract = "CIS2-wCCD",
+    name = "initialize",
+    parameter = "InitializeImplementationParams",
+    mutable
+)]
+fn contract_initialize<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<StateImplementation<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    ensure!(!host.state().initialized, concordium_cis2::Cis2Error::Custom(AlreadyInitialized));
+    // Set proxy and storage addresses.
+    let params: InitializeImplementationParams = ctx.parameter_cursor().get()?;
+    host.state_mut().proxy_address = params.proxy_address;
+    host.state_mut().state_address = params.state_address;
+    // Set the w_ccd_implementation contract as initialized.
+    host.state_mut().initialized = true;
```

Comment:
Same as for the other contract. You could just use `proxy_address.is_some()` to mean "initialized".